src/integrations/stripe.ts

```diff
@@ -162,6 +143,86 @@ export async function setupPaymentMethodAndPay(
   return null;
 }
 
+/**
+ * Used to either collect payment directly or launch stripe checkout to collect payment.
+ */
+export async function collectPayment(
+  actionData: ActionData,
+  stripePromise = getStripe()
+) {
+  const {
+    client,
+    triggerElement, // action button that triggered this
+    formattedFields,
+    updateFieldValues,
+    integrationData: stripeConfig
+  } = actionData;
+
+  // sync fields to BE
+  await syncFieldChanges(client, formattedFields, stripeConfig);
+
+  // setup any payment method on this step
+  if (actionData.targetElement) {
+    const payMethodResult = await setupPaymentMethod(
+      actionData,
+      false,
+      stripePromise
+    );
+    if (payMethodResult) {
+      // error
+      return payMethodResult;
+    }
+  }
+
+  // Now collect payment or do stripe checkout
+
+  try {
+    const stripe: any = await stripePromise;
+    if (stripeConfig.metadata.checkout_type === 'stripe') {
+      // stripe checkout
+      const {
+        success_url: successUrl = window.location.href,
+        cancel_url: cancelUrl = null
+      } = triggerElement?.properties || {};
+      const { checkout_url: checkoutUrl, error = null } =
+        await client.createCheckoutSession(successUrl, cancelUrl);
+      if (error) return { errorField: triggerElement, errorMessage: error };
+      // preserve browser back button function to current page/step
+      window.location.href = checkoutUrl;
+    } else {
```

Comment:
instead of a blanket else, IMO it's best practice to be explicit w/ an else-if even if you believe there's only one other else branch for now. that may be true now but may lead to issues in future development. also helps witgh understanding the code

Reply:
Actually, this was intentionally coded this way because legacy forms will not necessarily have have checkout_type (default is custom).  I will put the explicit check in along with the check of undefined, but it does seem messy to me.